model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/group/entity/JpaGroupEntity.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa.group.entity;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.persistence.Basic;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.persistence.Version;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+import org.hibernate.annotations.TypeDefs;
+import org.keycloak.models.map.common.DeepCloner;
+import org.keycloak.models.map.group.MapGroupEntity.AbstractGroupEntity;
+import static org.keycloak.models.map.storage.jpa.Constants.CURRENT_SCHEMA_VERSION_GROUP;
+import org.keycloak.models.map.storage.jpa.JpaRootEntity;
+import org.keycloak.models.map.storage.jpa.hibernate.jsonb.JsonbType;
+
+/**
+ * There are some fields marked by {@code @Column(insertable = false, updatable = false)}.
+ * Those fields are automatically generated by database from json field,
+ * therefore marked as non-insertable and non-updatable to instruct hibernate.
+ */
+@Entity
+@Table(name = "\"group\"", uniqueConstraints = {@UniqueConstraint(columnNames = {"realmId", "name", "parentId"})})
```

Comment:
Could a non-keyword be used for the table name, e.g. `KC_GROUP`? This would be less error prone both in liquibase scripts and potential migrations, and also could help the administrators who would be querying the database directly. Perhaps applies to roles as well?

Reply:
Adding a `KC_` prefix to this table would the mandate to add it to all other tables as well. Hm....

Reply:
Yup. I'm not arguing for a prefix per se, rather looking for a better name. I wouldn't like `GROUP_` either 😃
Perhaps `GROUPS`, but that would mandate using plural elsewhere as well.
Any better suggestions?

Reply:
Legacy jpa store uses `KEYCLOAK_GROUP` as well as `KEYCLOAK_ROLE` for roles. I wanted to leave it consistent with other stores, therefore the `\"group\"` was used. I can see reasons for not using it. Yet I'm not sure what would be the best name here.

Reply:
Let's then use `KC_` prefix everywhere - its short, and it seems that not only in my experience [singular is preferred](https://stackoverflow.com/questions/338156/table-naming-dilemma-singular-vs-plural-names).
It also has the benefit for clearly distinguishing keycloak tables from custom tables, should there be any.

Reply:
https://github.com/keycloak/keycloak/issues/10101

Reply:
resolved